### ISSUE TITLE
[Feature] Error Tracking via logs

### ISSUE THREAD

**Issue by TheLevti** [DataDog/dd-trace-php#2628]
### Describe the feature you'd like

Basically we are currently integrating error tracking via RUM, APM and LOGS. Because of APM being heavily sampled, we want to rely more on log error tracking.

Unfortunately the documentation is lacking support for error tracking via php https://docs.datadoghq.com/logs/error_tracking/backend/?tab=serilog#php

It would be really nice if the php-dd-trace extension is also able to support error tracking via logs. e.g. when using monolog and the errors/critical context has a Throwable object, make sure it gets picked up properly by the error tracking feature.

### Describe alternatives you've considered

I have also attempted to provide a custom formatter to set the specified generic fields `error.stack`, `error.message` and `error.kind`, but it seems like datadog does not convert/does not pick up those logs into the error tracking component.

**Comment by TheLevti** [comment]
Great! I was attempting something similar, instead I added a datadog processor that would populate those fields in the `extras` array to not pollute the given context from the application (if `context.exception` is provided of course).

What I did not do is adding remapping settings, but instead added my own custom JsonFormatter that would move the error array out of extras into the root. So it should have also worked, or?

I will try to do how you did it.

Would be nice if the default php remappers can check both, the extras and context array. Thank you for the hints, will let you know if it worked.

**Comment by adam-marchewicz-tg** [comment]
Are there any plans to support default monolog behaviour (`context.*`) so remappers won't be needed ?

**Comment by TheLevti** [comment]
> Are there any plans to support default monolog behaviour (`context.*`) so remappers won't be needed ?

Context should not be polluted with meta data. Thats what the extra collection is for. Context must be strictly related to the logged message.

**Comment by adam-marchewicz-tg** [comment]
Right, but even using `extra`, remapping is still required to make it work with dd error tracking.
